java/vector/src/test/java/org/apache/arrow/vector/TestValueVector.java

```diff
@@ -332,6 +332,204 @@ public void testSizeOfValueBuffer() {
+
+      /* vector data should be zeroed out */
+      for (int i = 0; i < capacityBeforeReset; i++) {
+        assertTrue("non-zero data not expected at index: " + i, floatVector.isNull(i));
+      }
+    }
+  }
+
+  @Test
+  public void testFixedFloat2WithPossibleTruncate() {
+    try (final Float2Vector floatVector = new Float2Vector(EMPTY_SCHEMA_PATH, allocator)) {
+      boolean error = false;
+      int initialCapacity = 16;
+
+      /* we should not throw exception for these values of capacity */
+      floatVector.setInitialCapacity(MAX_VALUE_COUNT - 1);
+      floatVector.setInitialCapacity(MAX_VALUE_COUNT);
+
+      try {
+        floatVector.setInitialCapacity(MAX_VALUE_COUNT * 4);
+      } catch (OversizedAllocationException oe) {
+        error = true;
+      } finally {
+        assertTrue(error);
+        error = false;
+      }
+
+      floatVector.setInitialCapacity(initialCapacity);
+      /* no memory allocation has happened yet so capacity of underlying buffer should be 0 */
+      assertEquals(0, floatVector.getValueCapacity());
+
+      /* allocate 32 bytes (16 * 2) */
+      floatVector.allocateNew();
+      /* underlying buffer should be able to store 16 values */
+      assertTrue(floatVector.getValueCapacity() >= initialCapacity);
+      initialCapacity = floatVector.getValueCapacity();
+
+      floatVector.zeroVector();
+
+      /* populate the floatVector */
+      floatVector.set(0, (short) 0x101c); // Float16.toFloat16(+0.00050163269043f)
+      floatVector.set(2, (short) 0x901c); // Float16.toFloat16(-0.00050163269043f)
+      floatVector.set(4, (short) 0x101d); // Float16.toFloat16(+0.000502109527588f)
+      floatVector.setWithPossibleTruncate(6, 2049.0f); // in f32=2049.000000, out f16=2048
+      floatVector.setWithPossibleTruncate(8, 4098.0f); // in f32=4098.000000, out f16=4096
+      floatVector.setWithPossibleTruncate(10, 8196.0f); // in f32=8196.000000, out f16=8192
+      floatVector.setWithPossibleTruncate(12, 16392.0f); // in f32=16392.000000, out f16=16384
+      floatVector.setWithPossibleTruncate(14, 32784.0f); // in f32=32784.000000, out f16=32768
+
+      try {
+        floatVector.setWithPossibleTruncate(initialCapacity, 65519.0f); // in f32=65519.000000, out f16=65504
+      } catch (IndexOutOfBoundsException ie) {
+        error = true;
+      } finally {
+        assertTrue(error);
+        error = false;
+      }
+
+      /* check vector contents */
+      assertEquals((short) 0x101c, floatVector.get(0));
+      assertEquals((short) 0x901c, floatVector.get(2));
+      assertEquals((short) 0x101d, floatVector.get(4));
+      assertEquals(2048.0f, floatVector.getValueAsFloat(6), 0);
+      assertEquals(4096.0f, floatVector.getValueAsFloat(8), 0);
+      assertEquals(8192.0f, floatVector.getValueAsFloat(10), 0);
+      assertEquals(16384.0f, floatVector.getValueAsFloat(12), 0);
+      assertEquals(32768.0f, floatVector.getValueAsFloat(14), 0);
```

Comment:
Can you please test `getValueAsDouble` as well?

Reply:
Added